bitcoin-rpc-provider/src/lib.rs

```diff
@@ -304,8 +313,16 @@ impl Wallet for BitcoinCoreProvider {
                         txid: x.txid,
                         vout: x.vout,
                     },
-                    address: x.address.as_ref().ok_or(Error::InvalidState)?.clone(),
-                    redeem_script: x.redeem_script.as_ref().unwrap_or(&Script::new()).clone(),
+                    address: x
+                        .address
+                        .as_ref()
+                        .map(|x| x.clone().assume_checked())
```

Comment:
We can and I thought of it. But then it'd be like asking the node who just sent you an address to also tell you its network. I guess it'd be a double check that the node is not doing something weird, but it doesn't feel very useful to me. I can add it though let me know if you think it's useful.

Reply:
I think it's low priority and I would only do this in a separate PR.

But I do think it can be useful in some spots. What if one side uses a testnet change address by mistake, when they're actually setting up a mainnet DLC?

Reply:
Which spots are you referring to? I think in this particular place, it wouldn't change anything. I think all the other places are in test code.

Reply:
You are right, this is the only place where we receive an `Address` in production code.

I assumed that there were more spots in the code base 😅 

> We can and I thought of it. But then it'd be like asking the node who just sent you an address to also tell you its network. I guess it'd be a double check that the node is not doing something weird, but it doesn't feel very useful to me.

Wouldn't it be more like making sure that the address returned by the node corresponds to the expected network? As in, you would receive the address and then check that its `Network` matches the one stored in `BitcoinCoreProvider`. This might prevent some weirdness/bugs if someone intends to run this component for `mainnet`, but they point to a `testnet` node instead.

Reply:
So IIUC what you're proposing is that the constructor takes a `Network` parameter? I think that's fine but I wonder if then it wouldn't be easier to just check within the constructor that the node corresponds to the expected network and then forget about it.